crates/storage/nippy-jar/src/compression/mod.rs

```diff
@@ -0,0 +1,66 @@
+use crate::NippyJarError;
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+
+mod zstd;
+pub use zstd::{Zstd, ZstdState};
+
+pub trait Compression {
+    /// Returns decompressed data.
+    fn decompress(&self, value: &[u8]) -> Result<Vec<u8>, NippyJarError>;
+
+    /// Compresses data from `src` to `dest`
+    fn compress_to<W: Write>(&self, src: &[u8], dest: &mut W) -> Result<(), NippyJarError>;
+
+    /// Returns `true` if it's ready to compress.
+    fn is_ready(&self) -> bool {
+        true
+    }
```

Comment:
what's the use case for this?
what's the caller supposed to do if it's not ready given that this is sync?